tests/test_jsvalidation.py

```
@@ -0,0 +1,419 @@
+    v1 = {
+        "type": "object",
+        "oneOf": [{
+            "type": "object",
+            "properties": {
+                "field_A": {"type": "string"}
+            }
+        }, {
+            "type": "object",
+            "properties": {
+                "field_A": {"type": "integer"}
+            }
+        }]
+    }
+    obj = JSONSchemaValidation()
+    with pytest.raises(JSONSchemaCompatibilityException):
+        obj.start(v1, 'first')
+
+
+def test_special_field_pass():
+    v1 = {
+        "type": "object",
+        "oneOf": [{
+            "type": "object",
+            "properties": {
+                "field_A": {"type": "string"}
+            }
+        }, {
+            "type": "object",
+            "properties": {
+                "field_B": {"type": "integer"}
+            }
+        }]
+    }
+    obj = JSONSchemaValidation()
+    obj.start(v1, 'first')
+
+
+def test_success_two_schemas():
+    v1 = {
+        "type": "object",
+        "properties": {
+            "field_A": {"type": "string"}
+        },
+        "required": ["field_A"]
+    }
+    v2 = {
+        "type": "object",
+        "properties": {
+            "field_B": {"type": "string"}
+        },
+        "required": ["field_B"]
+    }
+    obj = JSONSchemaValidation()
+    obj.start(v1, 'first')
+    obj.start(v2, 'second')
+
+
+def test_failure_between_schemas():
+    v1 = {
+        "type": "object",
+        "properties": {
+            "field_A": {"type": "string"}
+        },
+        "required": ["field_A"]
+    }
+    v2 = {
+        "type": "object",
+        "properties": {
+            "field_B": {"type": "string"}
+        },
+        "required": ["field_B"]
+    }
+    v3 = {
+        "type": "object",
+        "properties": {
+            "field_B": {"type": "string"},
+            "field_A": {"type": "number"}
+        },
+        "required": ["field_B"]
+    }
+    obj = JSONSchemaValidation()
+    with pytest.raises(JSONSchemaCompatibilityException):
```

Only the third one should raise the exception. The two first validation can be moved outside of this context.